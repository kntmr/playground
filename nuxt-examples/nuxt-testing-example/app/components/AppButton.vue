<template>
  <div>
    <p v-if="clicked">clicked</p>
    <button type="button" @click="click">click</button>
  </div>
</template>

<script>
export default {
  data () {
    return {
      clicked: false
    }
  },
  methods: {
    click () {
      this.clicked = true
    }
  }
}
</script>