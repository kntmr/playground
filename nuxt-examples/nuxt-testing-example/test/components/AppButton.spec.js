import AppButton from '~/components/AppButton.vue'
import { mount } from '@vue/test-utils'

describe('component/AppButton.vue', () => {
  test('snapshot 1', () => {
    const wrapper = mount(AppButton)
    expect(wrapper.element).toMatchSnapshot()
  })
  test('snapshot 2', () => {
    const wrapper = mount(AppButton)
    wrapper.find('button').trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })
})