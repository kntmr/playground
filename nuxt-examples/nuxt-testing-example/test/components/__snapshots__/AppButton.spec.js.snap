// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/AppButton.vue snapshot 1 1`] = `
<div>
  <!---->
   
  <button
    type="button"
  >
    click
  </button>
</div>
`;

exports[`component/AppButton.vue snapshot 2 1`] = `
<div>
  <!---->
   
  <button
    type="button"
  >
    click
  </button>
</div>
`;
